player/command.c

```diff
@@ -1055,13 +1080,18 @@ static int mp_property_list_chapters(void *ctx, struct m_property *prop,
         }
 
         for (n = 0; n < count; n++) {
+            if (n == cur) {
+                char *selected_style = get_selected_style(mpctx, res);
+                res = talloc_strdup_append(res, selected_style);
+                if (mpctx->video_out && mpctx->opts->video_osd)
+                    talloc_free(selected_style);
```

Comment:
How about instead `get` style, we do `append` style directly to `res`? So we avoid the need to keep track of the allocation and directly write to the result buffer?